### PR TITLE
fix: old pods are listed in new pod section for devtron apps deployed through Helm

### LINKED ISSUE #4024: Bug: old pods are listed in new pod section for devtron apps deployed through Helm
### 📜 Description

Old pods are listed in new pod section for devtron apps deployed through Helm. It should be listed in new pod section.
Screenshot 2023-10-05 at 6 05 45 PM

### 👟 Reproduction steps

1. Create a Devtron application.
2. Deploy it through Helm (non-gitops).
3. Do the configuration changes in pod spec and redeploy it.
4. New pod will be listed in New pod section but old pods will be still there and it won't move to old pod section.

### 👍 Expected behavior

Older pods should be listed in old pods section and not in new pod.

### 👎 Actual Behavior

old pods are listed in new pod section

### ☸ Kubernetes version

1.26

### Cloud provider

<details>

</details>

### 🌍 Browser

Chrome

### 🧱 Your Environment

_No response_

### ✅ Proposed Solution

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find any similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

AB#5985

### PR DESCRIPTION
**Pull request by ashishdevtron**
# Description
These changes fix the flag condition which decides whether the pods are new or old

Fixes #4024 

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: api/restHandler/AppListingRestHandler.go [hunk 1]
```diff
@@ -1594,6 +1594,14 @@ func (handler AppListingRestHandlerImpl) fetchResourceTree(w http.ResponseWriter
 		}
 		if detail != nil && detail.ReleaseExist {
 			resourceTree = util2.InterfaceToMapAdapter(detail.ResourceTreeResponse)
+			// add "isNew = false" flag in podMetaData - isNew flag will be omitted when the value is false
+			podMetadata := resourceTree["podMetadata"]
+			for _, data := range podMetadata.([]interface{}) {
+				dataMap := data.(map[string]interface{})
+				if _, ok := dataMap["isNew"]; !ok {
```
Comment (kartik-579) [inline api/restHandler/AppListingRestHandler.go:1601]: replace with constant

File: api/restHandler/AppListingRestHandler.go [hunk 2]
```diff
@@ -1594,6 +1594,14 @@ func (handler AppListingRestHandlerImpl) fetchResourceTree(w http.ResponseWriter
 		}
 		if detail != nil && detail.ReleaseExist {
 			resourceTree = util2.InterfaceToMapAdapter(detail.ResourceTreeResponse)
+			// add "isNew = false" flag in podMetaData - isNew flag will be omitted when the value is false
+			podMetadata := resourceTree["podMetadata"]
```
Comment (kartik-579) [inline api/restHandler/AppListingRestHandler.go:1598]: replace with constant

File: api/restHandler/AppListingRestHandler.go [hunk 3]
```diff
@@ -1594,6 +1594,14 @@ func (handler AppListingRestHandlerImpl) fetchResourceTree(w http.ResponseWriter
...
+			for _, data := range podMetadata.([]interface{}) {
```
Comment (kartik-579) [inline api/restHandler/AppListingRestHandler.go:1599]: is this type assertion always going to work? should add a validation for this

File: api/restHandler/AppListingRestHandler.go [hunk 4]
```diff
@@ -1594,6 +1594,14 @@ func (handler AppListingRestHandlerImpl) fetchResourceTree(w http.ResponseWriter
...
+				dataMap := data.(map[string]interface{})
```
Comment (kartik-579) [inline api/restHandler/AppListingRestHandler.go:1600]: check for assertion

**Review by kartik-579 — CHANGES_REQUESTED** [review]

**Comment by vikramdevtron** [comment]
Outdated, Not required anymore

**Pull request closed without merging**